clang error: abs is not constexpr

`FlexibleSUSY` doesn't compile with `clang` with a message

```
src/numerics.cpp:49:16: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
constexpr bool is_zero(double m, double tol) noexcept
               ^
src/numerics.cpp:51:22: note: non-constexpr function 'abs' cannot be used in a constant expression
   const double am = std::abs(m);
                     ^
/opt/local/libexec/llvm-7.0/include/c++/v1/math.h:791:1: note: declared here
abs(double __lcpp_x) _NOEXCEPT {return ::fabs(__lcpp_x);}
^
src/numerics.cpp:60:16: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
constexpr bool is_close(double m1, double m2, double tol) noexcept
               ^
src/numerics.cpp:64:33: note: non-constexpr function 'abs' cannot be used in a constant expression
   const double mmax = std::max(abs(m1), abs(m2));
                                ^
/opt/local/libexec/llvm-7.0/include/c++/v1/math.h:791:1: note: declared here
abs(double __lcpp_x) _NOEXCEPT {return ::fabs(__lcpp_x);}
^
```
Maybe we can replace `abs` with ternary if.

--- a/src/numerics.cpp
+++ b/src/numerics.cpp
@@ -42,13 +42,14 @@ namespace {
 constexpr double EPSTOL = 1.0e-11; ///< underflow accuracy
 constexpr double TOL = 1e-4;
 
+constexpr double dabs(double a) noexcept { return a >= 0. ? a : -a; }
 constexpr double sqr(double a) noexcept { return a*a; }
 constexpr double pow3(double a) noexcept { return a*a*a; }
 constexpr double pow6(double a) noexcept { return a*a*a*a*a*a; }
 
 constexpr bool is_zero(double m, double tol) noexcept
 {
-   const double am = std::abs(m);
+   const double am = dabs(m);
    const double mtol = tol * am;
 
    if (mtol == 0.0 && am != 0.0 && tol != 0.0)
@@ -59,10 +60,8 @@ constexpr bool is_zero(double m, double tol) noexcept
 
 constexpr bool is_close(double m1, double m2, double tol) noexcept
 {
-   using std::abs;
-
-   const double mmax = std::max(abs(m1), abs(m2));
-   const double mmin = std::min(abs(m1), abs(m2));
+   const double mmax = std::max(dabs(m1), dabs(m2));
+   const double mmin = std::min(dabs(m1), dabs(m2));
    const double max_tol = tol * mmax;
 
    if (max_tol == 0.0 && mmax != 0.0 && tol != 0.0)

